modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/PublicationUserNotificationHandlerTest.java

```diff
@@ -306,6 +312,260 @@ public void testGetLinkToViewConflicts() throws Exception {
+					TestPropsValues.getUserId());
+
+			for (UserNotificationEvent userNotificationEvent :
+					userNotificationEvents) {
+
+				if (!Objects.equals(
+						CTPortletKeys.PUBLICATIONS,
+						userNotificationEvent.getType())) {
+
+					continue;
+				}
+
+				JSONObject jsonObject = _jsonFactory.createJSONObject(
+					userNotificationEvent.getPayload());
+
+				if (jsonObject.getBoolean("isAdmin")) {
+					Assert.assertEquals(
+						_getAdminUserId(userNotificationEvent.getCompanyId()),
+						userNotificationEvent.getUserId());
+				}
+			}
+		}
+		finally {
+			if (journalServiceBundle != null) {
+				journalServiceBundle.start();
+			}
+		}
+	}
+
+	@Test
+	public void testGetUserNotificationEvent() throws Exception {
+		CTCollection ctCollection = _ctCollectionLocalService.addCTCollection(
+			null, TestPropsValues.getCompanyId(), TestPropsValues.getUserId(),
+			0, RandomTestUtil.randomString(), null);
+
+		try (SafeCloseable safeCloseable =
+				CTCollectionThreadLocal.setCTCollectionIdWithSafeCloseable(
+					ctCollection.getCtCollectionId())) {
+
+			JournalTestUtil.addArticle(
+				_group.getGroupId(), RandomTestUtil.randomString(),
+				StringPool.BLANK);
+		}
+
+		Bundle journalServiceBundle = null;
+
+		try {
+			Bundle bundle = FrameworkUtil.getBundle(
+				JournalArticleService.class);
+
+			BundleContext bundleContext = bundle.getBundleContext();
+
+			for (Bundle curBundle : bundleContext.getBundles()) {
+				if (Objects.equals(
+						curBundle.getSymbolicName(),
+						"com.liferay.journal.service") &&
+					(curBundle.getState() == Bundle.ACTIVE)) {
+
+					curBundle.stop();
+
+					journalServiceBundle = curBundle;
+
+					break;
+				}
+			}
+
+			_publishCTCollection(ctCollection.getCtCollectionId());
+
+			List<UserNotificationEvent> userNotificationEvents =
+				_userNotificationEventLocalService.getUserNotificationEvents(
+					TestPropsValues.getUserId());
+
+			for (UserNotificationEvent userNotificationEvent :
+					userNotificationEvents) {
+
+				if (!Objects.equals(
+						CTPortletKeys.PUBLICATIONS,
+						userNotificationEvent.getType())) {
+
+					continue;
+				}
+
+				JSONObject jsonObject = _jsonFactory.createJSONObject(
+					userNotificationEvent.getPayload());
+
+				if (jsonObject.getBoolean("isAdmin")) {
+					Assert.assertNotNull(userNotificationEvent);
+
+					Assert.assertFalse(jsonObject.getBoolean("showConflicts"));
+				}
+			}
+		}
+		finally {
+			if (journalServiceBundle != null) {
+				journalServiceBundle.start();
+			}
+		}
+	}
+
+	private long _getAdminUserId(long companyId) throws PortalException {
+		Role role = _roleLocalService.getRole(
+			companyId, RoleConstants.ADMINISTRATOR);
+
+		long[] userIds = _userLocalService.getRoleUserIds(role.getRoleId());
+
+		if (userIds.length == 0) {
+			throw new NoSuchUserException(
+				StringBundler.concat(
+					"No user exists in company ", companyId, " with role ",
+					role.getName()));
+		}
+
+		return userIds[0];
+	}
+
```

Comment:
Consolidate the test cases so that we don't need to do the same test setup multiple times.